Superbridge

scleakey opened the pull request:
- add superbridge class, tests, and documentation
- extract some functions from structure classes
- fix bug in structure log
- change `units.helpers._*` to `units._helpers.*`
- change copyright dates from 2024 to 2025

joe-pierce commented:
As discussed, please can you add some more tests to cover the following things:
- Test to change some attributes of the superbridge and write out to confirm changes are made. Maybe also to change to something invalid and assert that `._write()` fails
- Test reading and writing the IED containing the superbridge, currently this brings up an issue with the write output including non-strings
- Add 1 or more superbridges into some dat file and put them in the main test data folder so that they can be picked up by test_dat.py and test the read/write of the unit in the context of a DAT file

scleakey commented:
@joe-pierce I have addressed all your comments but I still need to deal with some new schema changes

scleakey commented:
I implemented schema changes, and now superbridge is part of `BRIDGE` with the subtype `INTEGRATED`. So I introduced a new attribute `BRIDGE.integrated_subtype` for the type of bridge in the integrated bridge.

joe-pierce commented:
Happy with all of this now 👍 Any future schema changes can be reflected in a new branch
